tests/test_mturk.py

```diff
@@ -30,132 +22,230 @@ class FixtureConfigurationError(Exception):
     """
 
 
+def system_marker():
+    # To prevent tests run on different systems trampling on each other,
+    # we mark data created in the MTurk sandbox with a value specific to
+    # each system.
+    return ':'.join(os.uname()).replace(' ', '')
+
+
 def name_with_hostname_prefix():
+    # Including the hostname in content created in the MTurk sandbox helps
+    # identify its source when reviewing records there.
     hostname = socket.gethostname()
     name = "{}:{}".format(hostname, generate_random_id(size=32))
     return name
 
 
-def as_resultset(things):
+def response_metadata():
+    # Most successful requests return an element like this in the JSON.
+    # In cases where there is no real return value, this is likely the only
+    # content in the response.
+    return {
+        'ResponseMetadata': {
+            'HTTPHeaders': {
+                'content-length': '123',
+                'content-type': 'application/x-amz-json-1.1',
+                'date': 'Thu, 08 Feb 2018 01:00:48 GMT',
+                'x-amzn-requestid': '806337c8-0c6b-11e8-9668-91a85782438a'
+            },
+            'HTTPStatusCode': 200,
+            'RequestId': '806337c8-0c6b-11e8-9668-91a85782438a',
+            'RetryAttempts': 0
+        }
+    }
+
+
+def as_batch_responses(key, things):
+    # Several MTurk calls return results in batches, with a "NextToken"
+    # indicating there is more content to be retrieved with a subsequent call.
     if not isinstance(things, (list, tuple)):
         things = [things]
-    result = ResultSet()
-    for thing in things:
-        result.append(thing)
-    return result
+
+    canned_response = [
+        {
+            u'NextToken': u'FAKE_NEXT_TOKEN',
+            u'NumResults': len(things),
+            key: things,
+            'ResponseMetadata': response_metadata()['ResponseMetadata']
+        },
+        {
+            u'NumResults': 0,
+            key: [],
+            'ResponseMetadata': response_metadata()['ResponseMetadata']
+        }
+    ]
+
+    return canned_response
 
 
 def fake_balance_response():
-    return as_resultset(Price(1.00))
+    return {
+        u'AvailableBalance': u'10000.00',
+        'ResponseMetadata': response_metadata()['ResponseMetadata']
+    }
```

Comment:
Why are we mixing string and unicode keys?

Reply:
It's a bit odd, but boto3 does this:
```
(Pdb) pp self.mturk.get_account_balance()
{u'AvailableBalance': u'10000.00',
 'ResponseMetadata': {'HTTPHeaders': {'content-length': '31',
                                      'content-type': 'application/x-amz-json-1.1',
                                      'date': 'Mon, 26 Feb 2018 00:18:58 GMT',
                                      'x-amzn-requestid': '2d01ab3b-f4dc-4128-9d24-4d2ac6611168'},
                      'HTTPStatusCode': 200,
                      'RequestId': '2d01ab3b-f4dc-4128-9d24-4d2ac6611168',
                      'RetryAttempts': 0}}
```

Reply:
Okay. I guess we'll see what happens on Python 3.